File: php/etl/Etl.php
<?php

declare(strict_types=1);

function transform(array $input): array
{
    $output = [];

    foreach($input as $points => $lettersArray) {
        foreach($lettersArray as $letter) {
            $output[strtolower($letter)] = $points; 
        }
    }

    return $output;
}

File: go/etl/etl.go
package etl

import "strings"
import "fmt"

func Transform(in map[int][]string) map[string]int {
	out := make(map[string]int)

	for points, lettersSlice := range in {
		for _, letter := range lettersSlice {
			out[strings.ToLower(letter)] = points
		}
	}

	return out
}
